internal/mode/static/nginx/config/servers_test.go

```diff
@@ -3052,33 +2984,199 @@ func TestGenerateProxySetHeaders(t *testing.T) {
 					Name:  "my-header",
 					Value: "",
 				},
+			}, grpcBaseHeaders...),
+			baseHeaders: grpcBaseHeaders,
+		},
```

Comment:
Do we have a test where keepalive checker returns true and the connection header is unset?

Reply:
no :/, do you think we should have that E2E test?

We currently have if keepalive checker returns true -> `getConnectionHeader` returns the unset connection header. 

Then we have, based off of what headers you pass into `generateProxySetHeaders` -> does generate the correct headers. (which kind of covers the -- the "connection header is unset")

Reply:
We are testing all the components, but it would be great to test that the location is created correctly when keep alive is enabled for an upstream. It kind of puts it all together.

Reply:
I added a test under `TestCreateServers` which I think covers this, let me know of your thoughts if otherwise!

Reply:
LGTM